src/Microsoft.VisualBasic/tests/VB/FileIOTestBase.vb

```diff
@@ -0,0 +1,148 @@
+            Dim testFileName As String = GenerateTestFileName(index, memberName, lineNumber)
+            Dim testFilePath As String = Path.Combine(TestDirectory, testFileName)
+
+            Const maxLength As Integer = 260 - 5 ' Windows MAX_PATH minus a bit
+
+            Dim excessLength As Integer = testFilePath.Length - maxLength
+
+            If excessLength > 0 Then
+                ' The path will be too long for Windows -- can we
+                ' trim memberName to fix it?
+                If excessLength < memberName.Length + "...".Length Then
+                    ' Take a chunk out of the middle as perhaps it's the least interesting part of the name
+                    memberName = memberName.Substring(0, memberName.Length \ 2 - excessLength \ 2) & "..." & memberName.Substring(memberName.Length \ 2 + excessLength \ 2)
+
+                    testFileName = GenerateTestFileName(index, memberName, lineNumber)
+                    testFilePath = Path.Combine(TestDirectory, testFileName)
+                Else
+                    Return fallbackGuid
+                End If
+            End If
+
+            Debug.Assert(testFilePath.Length <= maxLength + "...".Length)
+
+            Return testFileName
+        End Function
+
+        '''**************************************************************************
+        ''' ;RemoveEndingSeparator
+        ''' <summary>
+        ''' Removes all directory separators at the end of a path.
+        ''' </summary>
+        ''' <param name="Path">a full or relative path.</param>
+        ''' <returns>If Path is a root path, the same value. Otherwise, removes any directory separators at the end.</returns>
+        ''' <remarks>We decided not to return path with separators at the end (VsWhidbey 54741).</remarks>
+        Public Function RemoveEndingSeparator(ByVal Path As String) As String
+            If IO.Path.IsPathRooted(Path) Then
+                ' If the path is rooted, attempt to check if it is a root path.
+                ' Note: IO.Path.GetPathRoot: C: -> C:, C:\ -> C:\, \\myshare\mydir -> \\myshare\mydir
+                ' BUT \\myshare\mydir\ -> \\myshare\mydir!!! This function will remove the ending separator of
+                ' \\myshare\mydir\ as well. Do not use IsRoot here.
+                If Path.Equals(IO.Path.GetPathRoot(Path), StringComparison.OrdinalIgnoreCase) Then
+                    Return Path
+                End If
+            End If
+
+            ' Otherwise, remove all separators at the end.
+            Return Path.TrimEnd(IO.Path.DirectorySeparatorChar, IO.Path.AltDirectorySeparatorChar)
+        End Function
+
+        Private Shared Function GenerateTestFileName(index? As Integer, memberName As String, lineNumber As Integer) As String ' randomness to avoid collisions between derived test classes using same base method concurrently
```

Comment:
This seems the same as the base class, why not use the base implementation?

Reply:
Isn't this the base class?

Reply:
Isn't it on FileCleanupTestBase

Reply:
FileCleanupTestBase is must MustInherit so I just put a thin slick over it
```
    Public Class FileIOTestBase
        Inherits FileCleanupTestBase
        Sub New()
            MyBase.New
        End Sub
    End Class
```

I can't use the one in System.IO because I can't access it.

Reply:
It should be accessible as it's in an assembly that's automatically referenced by all tests. I went into Dummy.vb in the VB tests and added this:
```vb
Imports System.IO

Namespace Microsoft.VisualBasic.Tests 

    Public Class Dummy 
    Inherits FileCleanupTestBase
```
and it compiles.

Reply:
Some of the Functions/Properties (TestDirectoy, RemoveEndingSeparator and GetTestFilePath)  I need are Protected when I use  Inherits FileCleanupTestBase my first comment was overly broad.

Reply:
The access issue is I can't include System.IO because there are too make conflicts, I just needed to use IO.FileCleanupTestBase.